% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/plotMA-methods.R
\docType{methods}
\name{plotMA}
\alias{plotMA}
\alias{plotMA,DESeqResults-method}
\alias{plotMA,DESeqAnalysis-method}
\alias{plotMeanAverage}
\title{MA-plot: plot differences versus averages for high-throughput data}
\usage{
\S4method{plotMA}{DESeqResults}(object, genes = NULL,
  gene2symbol = NULL, ntop = 0L, direction = c("both", "up", "down"),
  pointColor = "gray50", sigPointColor = c(upregulated = "purple",
  downregulated = "orange"), return = c("ggplot", "DataFrame"))

\S4method{plotMA}{DESeqAnalysis}(object, results = 1L,
  lfcShrink = TRUE, genes = NULL, ntop = 0L, direction = c("both",
  "up", "down"), pointColor = "gray50", sigPointColor = c(upregulated =
  "purple", downregulated = "orange"), return = c("ggplot", "DataFrame"))

plotMeanAverage(...)
}
\arguments{
\item{object}{
    A data object, typically containing count values from an RNA-Seq experiment or microarray intensity values.
  }

\item{genes}{\code{character}. Gene identifiers. It is considered better practice
to input the stable gene identifiers from Ensembl (e.g. "ENSG00000000003")
and not the (HGNC) gene symbols (e.g. "TSPN6"), if possible.}

\item{gene2symbol}{\code{Gene2Symbol}. Gene-to-symbol mappings. Must contain
\code{geneID} and \code{geneName} columns. See \code{Gene2Symbol()} for more information.}

\item{ntop}{\code{scalar integer}. Number of top genes to label.}

\item{direction}{\code{string}. Plot "\code{both}", "\code{up}", or "\code{down}" directions.}

\item{pointColor}{\code{string}. Default point color for the plot.}

\item{sigPointColor}{\code{character}. Color names for labeling upregulated and
downregulated genes. Also supports a character string for labeling DEGs
with the same color, regardless of direction.}

\item{return}{\code{string}. Return type. Uses \code{base::match.arg()}
internally and defaults to the first argument in the \code{character} vector.}

\item{results}{\code{scalar}. Position or name of \code{DESeqResults}.}

\item{lfcShrink}{\code{boolean}. Use shrunken log2 fold change (LFC) values.}

\item{...}{
    Additional arguments, for use in specific methods.
  }
}
\value{
\code{ggplot}.
}
\description{
A generic function which produces an MA-plot for an object containing microarray, RNA-Seq
  or other data.
}
\details{
An MA plot is an application of a Blandâ€“Altman plot for visual representation
of genomic data. The plot visualizes the differences between measurements
taken in two samples, by transforming the data onto M (log ratio) and A
(mean average) scales, then plotting these values.
}
\note{
We are not allowing post hoc \code{alpha} or \code{lfcThreshold} cutoffs here.
}
\examples{
data(deseq)

dds <- as(deseq, "DESeqDataSet")
g2s <- Gene2Symbol(dds)
geneIDs <- head(g2s[["geneID"]])
print(geneIDs)
geneNames <- head(g2s[["geneName"]])
print(geneNames)

## DESeqAnalysis ====
plotMA(deseq)

## Customize the colors.
plotMA(
    object = deseq,
    pointColor = "black",
    sigPointColor = "purple"
)
plotMA(
    object = deseq,
    sigPointColor = c(
        upregulated = "green",
        downregulated = "red"
    )
)

## Directional support (up or down).
plotMA(deseq, direction = "up", ntop = 5L)
plotMA(deseq, direction = "down", ntop = 5L)

## Label genes manually.
## Note that either gene IDs or names (symbols) are supported.
plotMA(deseq, genes = geneIDs)
plotMA(deseq, genes = geneNames)
}
\seealso{
\code{DESeq2::plotMA()}.
}
\author{
Michael Steinbaugh, Rory Kirchner
}
