---
params:
  title: "Differential expression analysis using DESeq2"
  
  # Use DESeqAnalysis object as input.
  # Basic: Load from file on disk, either RDS or RDA.
  # Advanced: Loop across multiple objects using `rmarkdown::render()`.
  object: "rds/YYYY-MM-DD/DESeqAnalysis.rds"
  # `name` param is only necessary when passing in object directly.
  name: !r NULL
  
  data_dir: !r file.path("data", Sys.Date())
  output_dir: !r file.path("results", Sys.Date(), "differential_expression")

title: "`r params$title`"
author: "`r getOption('author')`"
date: "`r Sys.Date()`"
bibliography: bibliography.bib
---

```{r setup, cache=FALSE, message=FALSE}
# Last modified 2019-03-11
library(DESeqAnalysis)
prepareTemplate()
source("_setup.R")
```

```{r header, child="_header.Rmd"}
```

```{r object}
if (is.character(params$object)) {
    if (fileExt(object) == "rds") {
        object <- readRDS(file = params$object)
        name <- basenameSansExt(params$object)
    } else if (fileExt(object) == "rda") {
        name <- load(params$object)
        object <- get(name, inherits = FALSE)
    }
} else {
    object <- params$object
    name <- params$name
}
stopifnot(
    is(object, "DESeqAnalysis"),
    isString(name)
)
invisible(validObject(object))
print(object)
```

# Sample data

```{r sample_data}
sampleData(object)
```

# MA plot {.tabset}

An MA plot compares transformed counts on `M` (log ratio) and `A` (mean average) scales [@Yang2002-sx].

```{r plot_ma}
invisible(lapply(
    X = resultsNames(object),
    object = object,
    FUN = function(results, object) {
        plotMA(
            object = object,
            results = results,
            lfcShrink = FALSE
        ) %>% print()
        plotMA(
            object = object,
            results = results,
            lfcShrink = TRUE
        ) %>% print()
    }
))
```

# Volcano plot {.tabset}

A volcano plot compares significance (BH-adjusted *P* value) against fold change (log2) [@Cui2003-rn; @Li2014-ll]. Genes in the green box with text labels have an adjusted *P* value are likely to be the top candidate genes of interest.

```{r plot_volcano}
invisible(lapply(
    X = resultsNames(object),
    object = object,
    FUN = function(results, object) {
        plotVolcano(
            object = object,
            results = results,
            lfcShrink = TRUE
        ) %>% print()
    }
))
```

# PCA {.tabset}

```{r plot_deg_pca}
invisible(lapply(
    X = resultsNames(object),
    object = object,
    FUN = function(results, object) {
        plotDEGPCA(
            object = object,
            results = results,
            contrastSamples = TRUE
        ) %>% print()
    }
))
```

# Heatmap {.tabset}

This plot shows only differentially expressed genes on a per-sample basis. We have scaled the data by row (z-score) and used the `ward.D2` method for clustering [@Ward1963-xf].

```{r plot_deg_heatmap}
invisible(lapply(
    X = resultsNames(object),
    object = object,
    FUN = function(results, object) {
        plotDEGHeatmap(
            object = object,
            results = results,
            contrastSamples = TRUE
        )
    }
))
```

# Top tables {.tabset}

Only the top up- and down-regulated genes (arranged by log2 fold change) are shown.

```{r top_tables, results="asis"}
invisible(lapply(
    X = resultsNames(object),
    FUN = function(results, object) {
        topTables(object, results = results)
    },
    object = object
))
```

# Export

Here we are exporting the data stored inside our `DESeqDataSet`, `DESeqTransform`, and `DESeqResults` objects (results = unshrunken LFC; lfcShrink = shrunken LFC), along with the differentially expressed gene (DEG) results tables (resultsTables).

The DEG results tables are sorted by BH-adjusted P value (except for the "all" file, which is unmodified), and contain the following columns:

- `baseMean`: Mean of the normalized counts per gene for all samples.
- `log2FoldChange`: log2 fold change.
- `lfcSE`: log2 standard error.
- `stat`: Wald statistic.
- `pvalue`: Walt test *P* value.
- `padj`: BH adjusted Wald test *P* value (corrected for multiple comparisons; aka FDR).

These files also contain genome annotation information and the size-factor adjusted normalized counts stored inside the `DESeqDataSet` used to perform the differential expression analysis.

Note that `colData` files contain annotations corresponding to the columns (i.e. samples) and the `rowData` files contain annotations corresponding to the rows (i.e. genes). You can obtain gene-to-symbol mappings using this `rowData` file, for example.

```{r export}
export(
    x = object,
    name = name,
    dir = params$output_dir,
    compress = FALSE
)
```

```{r footer, child="_footer.Rmd"}
```
