<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html>
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>DESeq2 Differential Expression Results Tables — DESeqResultsTables • DESeqAnalysis</title>

<!-- jquery -->
<script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="DESeq2 Differential Expression Results Tables — DESeqResultsTables" />

<meta property="og:description" content="DESeqResults object with DataFrame subsets and file path information." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src='https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML'></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DESeqAnalysis</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/steinbaugh/DESeqAnalysis/">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>DESeq2 Differential Expression Results Tables</h1>
    <small class="dont-index">Source: <a href='https://github.com/steinbaugh/DESeqAnalysis//blob/master/R/AllGenerics.R'><code>R/AllGenerics.R</code></a>, <a href='https://github.com/steinbaugh/DESeqAnalysis//blob/master/R/DESeqResultsTables-methods.R'><code>R/DESeqResultsTables-methods.R</code></a></small>
    <div class="hidden name"><code>DESeqResultsTables.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p><code>DESeqResults</code> object with <code>DataFrame</code> subsets and file path information.</p>
    
    </div>

    <pre class="usage"><span class='fu'>DESeqResultsTables</span>(<span class='no'>object</span>, <span class='no'>...</span>)

<span class='co'># S4 method for DESeqResults</span>
<span class='fu'>DESeqResultsTables</span>(<span class='no'>object</span>)

<span class='co'># S4 method for DESeqAnalysis</span>
<span class='fu'>DESeqResultsTables</span>(<span class='no'>object</span>, <span class='kw'>results</span> <span class='kw'>=</span> <span class='fl'>1L</span>,
  <span class='kw'>lfcShrink</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>Object.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments.</p></td>
    </tr>
    <tr>
      <th>results</th>
      <td><p><code>scalar</code>. Position or name of <code>DESeqResults</code>.</p></td>
    </tr>
    <tr>
      <th>lfcShrink</th>
      <td><p><code>boolean</code>. Use shrunken log2 fold change (LFC) values.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>DESeqResultsTables</code>, containing the original, unmodified
<code>DESeqResults</code> <code>DataFrame</code> along with the corresponding differentially
expressed genes and gene-level metadata (rowRanges).</p>
    
    <h2 class="hasAnchor" id="obtaining-results-from-deseq-"><a class="anchor" href="#obtaining-results-from-deseq-"></a>Obtaining results from DESeq2</h2>

    
    <p>It is recommended to specify the <code>contrast</code> argument as <code>character</code>:</p><ol>
<li><p>Design matrix factor of interest.</p></li>
<li><p>Numerator for LFC (expt).</p></li>
<li><p>Denominator for LFC (control).</p></li>
</ol>
    <p>For example, to get the relative change in gene expression between mutant
and wildtype samples, here's how to set the contrast vector:</p><pre>c(
    factor = "genotype",
    numerator = "mutant",
    denominator = "wildtype"
)
</pre>
    
    <h2 class="hasAnchor" id="log-fold-change-threshold-cutoffs"><a class="anchor" href="#log-fold-change-threshold-cutoffs"></a>Log fold change threshold cutoffs</h2>

    
    <p>Refer to <code><a href='http://www.rdocumentation.org/packages/DESeq2/topics/results'>DESeq2::results()</a></code> for additional information about using
<code>lfcThreshold</code> and <code>altHypothesis</code> to set an alternative hypothesis based on
expected fold changes. In addition, the "Hypothesis tests with thresholds on
effect size" section in the DESeq2 paper provides additional explanation.</p>
<p>Don't apply <em>post hoc</em> LFC threshold filtering to obtain results tables, as
this will destroy the meaning of the adjusted <em>P</em> values computed. If you are
expecting a biological effect size past a particular threshold, set
<code>lfcThreshold</code>, but be conservative.</p>
<p>This <a href='https://support.bioconductor.org/p/101504/'>thread</a> on the Bioconductor forums explains how <code><a href='http://www.rdocumentation.org/packages/DESeq2/topics/results'>DESeq2::results()</a></code>
should be called with regard to LFC cutoffs in nice detail. In particular,
refer to Mike Love:</p>
<p>"A common procedure is to disregard genes whose estimated LFC <em>beta ir</em> is
below some threshold, <em>beta ir ≤ theta</em>. However, this approach loses the
benefit of an easily interpretable FDR, as the reported <em>P</em> value and
adjusted <em>P</em> value still correspond to the test of <em>zero</em> LFC. It is
therefore desirable to include the threshold in the statistical testing
procedure directly, i.e., not to filter post hoc on a reported fold-change
<em>estimate</em>, but rather to evaluate statistically directly whether there is
sufficient evidence that the LFC is above the chosen threshold."</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><ul>
<li><p><code><a href='http://www.rdocumentation.org/packages/DESeq2/topics/results'>DESeq2::results()</a></code>.</p></li>
<li><p><code>markdown()</code>, <code>write()</code>.</p></li>
</ul></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>data</span>(<span class='no'>deseq</span>)

<span class='co'>## DESeqAnalysis ====</span>
<span class='co'>## This is the recommended default method.</span>
<span class='no'>x</span> <span class='kw'>&lt;-</span> <span class='fu'>DESeqResultsTables</span>(<span class='no'>deseq</span>)</div><div class='output co'>#&gt; <span class='message'>DESeqResults: treatment folic acid vs control (shrunken LFC)</span></div><div class='input'><span class='fu'>print</span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; DESeqResultsTables 1.22.1
#&gt; contrast: treatment folic acid vs control
#&gt; alpha: 0.1
#&gt; lfcThreshold: 0
#&gt; out of 500 with nonzero total read count
#&gt; adjusted p-value &lt; 0.1
#&gt; LFC &gt; 0 (up)       : 115, 23%
#&gt; LFC &lt; 0 (down)     : 139, 28%
#&gt; outliers [1]       : 6, 1.2%
#&gt; low counts [2]     : 0, 0%
#&gt; (mean count &lt; 33)</div><div class='input'>
<span class='co'>## DESeqResults ====</span>
<span class='no'>res</span> <span class='kw'>&lt;-</span> <span class='fu'>as</span>(<span class='no'>deseq</span>, <span class='st'>"DESeqResults"</span>)
<span class='no'>x</span> <span class='kw'>&lt;-</span> <span class='fu'>DESeqResultsTables</span>(<span class='no'>res</span>)
<span class='fu'>print</span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; DESeqResultsTables 
#&gt; contrast: treatment folic acid vs control
#&gt; alpha: 0.1
#&gt; lfcThreshold: 0
#&gt; out of 500 with nonzero total read count
#&gt; adjusted p-value &lt; 0.1
#&gt; LFC &gt; 0 (up)       : 115, 23%
#&gt; LFC &lt; 0 (down)     : 139, 28%
#&gt; outliers [1]       : 6, 1.2%
#&gt; low counts [2]     : 0, 0%
#&gt; (mean count &lt; 33)</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#obtaining-results-from-deseq-">Obtaining results from DESeq2</a></li>

      <li><a href="#log-fold-change-threshold-cutoffs">Log fold change threshold cutoffs</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='/'>Michael Steinbaugh</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
   </div>

  

  </body>
</html>

