<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>DESeq2 differential expression analysis — DESeqAnalysis-class • DESeqAnalysis</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="DESeq2 differential expression analysis — DESeqAnalysis-class" />

<meta property="og:description" content="Class containing all elements generated during differential expression
analysis with DESeq2. This class is essentially a list with validity checks
to ensure DESeqTransform and DESeqResults correspond to the
DESeqDataSet." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DESeqAnalysis</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.16</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/acidgenomics/DESeqAnalysis">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>DESeq2 differential expression analysis</h1>
    <small class="dont-index">Source: <a href='https://github.com/acidgenomics/DESeqAnalysis/blob/master/R/AllClasses.R'><code>R/AllClasses.R</code></a>, <a href='https://github.com/acidgenomics/DESeqAnalysis/blob/master/R/DESeqAnalysis.R'><code>R/DESeqAnalysis.R</code></a></small>
    <div class="hidden name"><code>DESeqAnalysis-class.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Class containing all elements generated during differential expression
analysis with DESeq2. This class is essentially a <code>list</code> with validity checks
to ensure <code>DESeqTransform</code> and <code>DESeqResults</code> correspond to the
<code>DESeqDataSet</code>.</p>
    
    </div>

    <pre class="usage"><span class='fu'>DESeqAnalysis</span>(<span class='no'>data</span>, <span class='no'>transform</span>, <span class='no'>results</span>, <span class='no'>lfcShrink</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>data</th>
      <td><p><code>DESeqDataSet</code>.</p></td>
    </tr>
    <tr>
      <th>transform</th>
      <td><p><code>DESeqTransform</code>.
<code><a href='https://www.rdocumentation.org/packages/DESeq2/topics/varianceStabilizingTransformation'>DESeq2::varianceStabilizingTransformation()</a></code> recommended by default.</p></td>
    </tr>
    <tr>
      <th>results</th>
      <td><p><code>list</code> or single <code>DESeqResults</code>.
One or more unshrunken <code>DESeqResults</code>.
Assign the <code><a href='https://www.rdocumentation.org/packages/DESeq2/topics/results'>DESeq2::results()</a></code> return here.</p></td>
    </tr>
    <tr>
      <th>lfcShrink</th>
      <td><p><code>list</code>or single <code>DESeqResults</code>.
One or more shrunken <code>DESeqResults</code>.
Assign the <code><a href='https://www.rdocumentation.org/packages/DESeq2/topics/lfcShrink'>DESeq2::lfcShrink()</a></code> return here.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>DESeqAnalysis</code>.
Contains a <code>DESeqDataSet</code>, <code>DESeqTransform</code>, and corresponding
<code>DESeqResults</code> list.</p>
    
    <h2 class="hasAnchor" id="slots"><a class="anchor" href="#slots"></a>Slots</h2>

    
    <dl class='dl-horizontal'>
<dt><code>data</code></dt><dd><p><code>DESeqDataSet</code>.</p></dd>
    <dt><code>transform</code></dt><dd><p><code>DESeqTransform</code>.</p></dd>
    <dt><code>results</code></dt><dd><p><code>list</code>.
One or more unshrunken <code>DESeqResults</code>.</p></dd>
    <dt><code>lfcShrink</code></dt><dd><p><code>list</code>.
<em>Optional</em>. One or more shrunken <code>DESeqResults</code>. If set, must correspond to
those defined in <code>results</code>.</p></dd>
</dl>
    
    <h2 class="hasAnchor" id="deseqdataset"><a class="anchor" href="#deseqdataset"></a>DESeqDataSet</h2>

    
    <p>We recommend generating the <code>DESeqDataSet</code> by coercion from <code>bcbioRNASeq</code>
object using <code><a href='as.html'>as(dds, "bcbioRNASeq")</a></code>. Don't use the <code><a href='https://www.rdocumentation.org/packages/DESeq2/topics/DESeqDataSet'>DESeq2::DESeqDataSet()</a></code>
or <code><a href='https://www.rdocumentation.org/packages/DESeq2/topics/DESeqDataSet'>DESeq2::DESeqDataSetFromMatrix()</a></code> constructors to generate the
<code>DESeqDataSet</code> object.</p>
    
    <h2 class="hasAnchor" id="deseqtransform"><a class="anchor" href="#deseqtransform"></a>DESeqTransform</h2>

    
    <p>Object containing variance-stabilized counts. We recommend slotting the
return from either <code><a href='https://www.rdocumentation.org/packages/DESeq2/topics/varianceStabilizingTransformation'>DESeq2::varianceStabilizingTransformation()</a></code> or
<code><a href='https://www.rdocumentation.org/packages/DESeq2/topics/rlog'>DESeq2::rlog()</a></code>.</p>
    
    <h2 class="hasAnchor" id="deseqresults"><a class="anchor" href="#deseqresults"></a>DESeqResults</h2>

    
    <p>Don't modify any of the <code>DESeqResults</code> objects manually. This includes
rearranging the rows or dropping genes without adjusted P values. We'll take
care of this automatically in supported methods.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><ul>
<li><p><code><a href='https://www.rdocumentation.org/packages/utils/topics/help'>help(topic = "Annotated-class", package = "S4Vectors")</a></code>.</p></li>
<li><p><code>showClass("Annotated")</code>.</p></li>
</ul></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/library'>library</a></span>(<span class='no'>DESeq2</span>)

<span class='no'>data</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/DESeq2/topics/DESeq'>DESeq</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/DESeq2/topics/makeExampleDESeqDataSet'>makeExampleDESeqDataSet</a></span>())</div><div class='output co'>#&gt; <span class='message'>estimating size factors</span></div><div class='output co'>#&gt; <span class='message'>estimating dispersions</span></div><div class='output co'>#&gt; <span class='message'>gene-wise dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>mean-dispersion relationship</span></div><div class='output co'>#&gt; <span class='message'>final dispersion estimates</span></div><div class='output co'>#&gt; <span class='message'>fitting model and testing</span></div><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/class'>class</a></span>(<span class='no'>data</span>)</div><div class='output co'>#&gt; [1] "DESeqDataSet"
#&gt; attr(,"package")
#&gt; [1] "DESeq2"</div><div class='input'>
<span class='no'>transform</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/DESeq2/topics/varianceStabilizingTransformation'>varianceStabilizingTransformation</a></span>(<span class='no'>data</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/class'>class</a></span>(<span class='no'>transform</span>)</div><div class='output co'>#&gt; [1] "DESeqTransform"
#&gt; attr(,"package")
#&gt; [1] "DESeq2"</div><div class='input'>
<span class='fu'><a href='resultsNames.html'>resultsNames</a></span>(<span class='no'>data</span>)</div><div class='output co'>#&gt; [1] "Intercept"        "condition_B_vs_A"</div><div class='input'><span class='no'>name</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='resultsNames.html'>resultsNames</a></span>(<span class='no'>data</span>)<span class='kw'>[[</span><span class='fl'>2L</span>]]
<span class='no'>results</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='results.html'>results</a></span>(<span class='no'>data</span>, <span class='kw'>name</span> <span class='kw'>=</span> <span class='no'>name</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/class'>class</a></span>(<span class='no'>results</span>)</div><div class='output co'>#&gt; [1] "DESeqResults"
#&gt; attr(,"package")
#&gt; [1] "DESeq2"</div><div class='input'>
<span class='no'>lfcShrink</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/DESeq2/topics/lfcShrink'>lfcShrink</a></span>(<span class='kw'>dds</span> <span class='kw'>=</span> <span class='no'>data</span>, <span class='kw'>res</span> <span class='kw'>=</span> <span class='no'>results</span>, <span class='kw'>coef</span> <span class='kw'>=</span> <span class='fl'>2L</span>)</div><div class='output co'>#&gt; <span class='message'>using 'normal' for LFC shrinkage, the Normal prior from Love et al (2014).</span>
#&gt; <span class='message'>additional priors are available via the 'type' argument, see ?lfcShrink for details</span></div><div class='input'>
<span class='no'>results</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a></span>(<span class='no'>results</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/names'>names</a></span>(<span class='no'>results</span>) <span class='kw'>&lt;-</span> <span class='no'>name</span>
<span class='no'>lfcShrink</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/list'>list</a></span>(<span class='no'>lfcShrink</span>)
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/names'>names</a></span>(<span class='no'>lfcShrink</span>) <span class='kw'>&lt;-</span> <span class='no'>name</span>
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/identical'>identical</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/names'>names</a></span>(<span class='no'>results</span>), <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/names'>names</a></span>(<span class='no'>lfcShrink</span>))</div><div class='output co'>#&gt; [1] TRUE</div><div class='input'>
<span class='no'>x</span> <span class='kw'>&lt;-</span> <span class='fu'>DESeqAnalysis</span>(
    <span class='kw'>data</span> <span class='kw'>=</span> <span class='no'>data</span>,
    <span class='kw'>transform</span> <span class='kw'>=</span> <span class='no'>transform</span>,
    <span class='kw'>results</span> <span class='kw'>=</span> <span class='no'>results</span>,
    <span class='kw'>lfcShrink</span> <span class='kw'>=</span> <span class='no'>lfcShrink</span>
)
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/print'>print</a></span>(<span class='no'>x</span>)</div><div class='output co'>#&gt; DESeqAnalysis 0.1.16; DESeq2 1.22.2
#&gt; data:
#&gt;   dim: 1000 12 
#&gt;   metadata(1): version
#&gt;   assays(4): counts mu H cooks
#&gt;   rownames(1000): gene1 gene2 ... gene999 gene1000
#&gt;   rowData names(25): trueIntercept trueBeta ... deviance maxCooks
#&gt;   colnames(12): sample1 sample2 ... sample11 sample12
#&gt;   colData names(2): condition sizeFactor
#&gt; transform: varianceStabilizingTransformation
#&gt; results: condition_B_vs_A
#&gt; lfcShrink: normal</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#slots">Slots</a></li>

      <li><a href="#deseqdataset">DESeqDataSet</a></li>

      <li><a href="#deseqtransform">DESeqTransform</a></li>

      <li><a href="#deseqresults">DESeqResults</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Michael Steinbaugh</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://acidgenomics.com/'>Acid Genomics</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

