<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>MA-plot: plot differences versus averages for high-throughput data — plotMA • DESeqAnalysis</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  
<meta property="og:title" content="MA-plot: plot differences versus averages for high-throughput data — plotMA" />

<meta property="og:description" content="A generic function which produces an MA-plot for an object containing microarray, RNA-Seq
  or other data." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DESeqAnalysis</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.20</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/acidgenomics/DESeqAnalysis">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>MA-plot: plot differences versus averages for high-throughput data</h1>
    <small class="dont-index">Source: <a href='https://github.com/acidgenomics/DESeqAnalysis/blob/master/R/plotMA-methods.R'><code>R/plotMA-methods.R</code></a></small>
    <div class="hidden name"><code>plotMA.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>A generic function which produces an MA-plot for an object containing microarray, RNA-Seq
  or other data.</p>
    
    </div>

    <pre class="usage"><span class='fu'>plotMA</span>(<span class='no'>object</span>, <span class='no'>...</span>)

<span class='co'># S4 method for DESeqResults</span>
<span class='fu'>plotMA</span>(<span class='no'>object</span>, <span class='kw'>genes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>gene2symbol</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>ntop</span> <span class='kw'>=</span> <span class='fl'>0L</span>, <span class='kw'>direction</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"both"</span>, <span class='st'>"up"</span>, <span class='st'>"down"</span>),
  <span class='kw'>pointColor</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='kw'>downregulated</span> <span class='kw'>=</span> <span class='st'>"darkorchid3"</span>, <span class='kw'>upregulated</span> <span class='kw'>=</span>
  <span class='st'>"darkorange2"</span>, <span class='kw'>nonsignificant</span> <span class='kw'>=</span> <span class='st'>"gray50"</span>), <span class='kw'>pointSize</span> <span class='kw'>=</span> <span class='fl'>2L</span>,
  <span class='kw'>pointAlpha</span> <span class='kw'>=</span> <span class='fl'>0.7</span>, <span class='kw'>return</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"ggplot"</span>, <span class='st'>"DataFrame"</span>))

<span class='co'># S4 method for DESeqAnalysis</span>
<span class='fu'>plotMA</span>(<span class='no'>object</span>, <span class='no'>results</span>, <span class='kw'>lfcShrink</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>genes</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>ntop</span> <span class='kw'>=</span> <span class='fl'>0L</span>, <span class='kw'>direction</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"both"</span>, <span class='st'>"up"</span>, <span class='st'>"down"</span>),
  <span class='kw'>pointColor</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='kw'>downregulated</span> <span class='kw'>=</span> <span class='st'>"darkorchid3"</span>, <span class='kw'>upregulated</span> <span class='kw'>=</span>
  <span class='st'>"darkorange2"</span>, <span class='kw'>nonsignificant</span> <span class='kw'>=</span> <span class='st'>"gray50"</span>), <span class='kw'>pointSize</span> <span class='kw'>=</span> <span class='fl'>2L</span>,
  <span class='kw'>pointAlpha</span> <span class='kw'>=</span> <span class='fl'>0.7</span>, <span class='kw'>return</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"ggplot"</span>, <span class='st'>"DataFrame"</span>))

<span class='fu'>plotMA2</span>(<span class='no'>object</span>, <span class='no'>...</span>)

<span class='co'># S4 method for DESeqDataSet</span>
<span class='fu'>plotMA2</span>(<span class='no'>object</span>, <span class='kw'>alpha</span> <span class='kw'>=</span> <span class='fl'>0.1</span>, <span class='kw'>main</span> <span class='kw'>=</span> <span class='st'>""</span>,
  <span class='kw'>xlab</span> <span class='kw'>=</span> <span class='st'>"mean of normalized counts"</span>, <span class='no'>ylim</span>, <span class='kw'>MLE</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='no'>...</span>)

<span class='co'># S4 method for DESeqResults</span>
<span class='fu'>plotMA2</span>(<span class='no'>object</span>, <span class='no'>alpha</span>, <span class='kw'>main</span> <span class='kw'>=</span> <span class='st'>""</span>,
  <span class='kw'>xlab</span> <span class='kw'>=</span> <span class='st'>"mean of normalized counts"</span>, <span class='no'>ylim</span>, <span class='kw'>MLE</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='no'>...</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>A data object, typically containing count values from an RNA-Seq experiment or microarray intensity values.</p></td>
    </tr>
    <tr>
      <th>genes</th>
      <td><p><code>character</code>.
Gene identifiers. It is considered better practice to input the stable gene
identifiers from Ensembl (e.g. "ENSG00000000003") and not the (HGNC) gene
symbols (e.g. "TSPN6"), if possible.</p></td>
    </tr>
    <tr>
      <th>gene2symbol</th>
      <td><p><code>Gene2Symbol</code>.
Gene-to-symbol mappings. Must contain <code>geneID</code> and <code>geneName</code> columns. See
<code>Gene2Symbol</code> for more information.</p></td>
    </tr>
    <tr>
      <th>ntop</th>
      <td><p><code><a href='https://www.rdocumentation.org/packages/base/topics/integer'>integer(1)</a></code>.
Number of top genes to label.</p></td>
    </tr>
    <tr>
      <th>direction</th>
      <td><p><code><a href='https://www.rdocumentation.org/packages/base/topics/character'>character(1)</a></code>.
Plot "<code>both</code>", "<code>up</code>", or "<code>down</code>" directions.</p></td>
    </tr>
    <tr>
      <th>pointColor</th>
      <td><p><code><a href='https://www.rdocumentation.org/packages/base/topics/character'>character(1)</a></code>.
Default point color for the plot.</p></td>
    </tr>
    <tr>
      <th>pointSize</th>
      <td><p><code><a href='https://www.rdocumentation.org/packages/base/topics/numeric'>numeric(1)</a></code>.
Point size. In the range of 1-3 is generally recommended.</p></td>
    </tr>
    <tr>
      <th>pointAlpha</th>
      <td><p><code><a href='https://www.rdocumentation.org/packages/base/topics/numeric'>numeric(1)</a></code>.
Alpha transparency level. Must be a proportion (0-1).</p></td>
    </tr>
    <tr>
      <th>return</th>
      <td><p><code><a href='https://www.rdocumentation.org/packages/base/topics/character'>character(1)</a></code>.
Return type. Uses <code><a href='https://www.rdocumentation.org/packages/base/topics/match.arg'>match.arg()</a></code> internally and defaults
to the first argument in the <code>character</code> vector.</p></td>
    </tr>
    <tr>
      <th>results</th>
      <td><p><code><a href='https://www.rdocumentation.org/packages/base/topics/character'>character(1)</a></code> or <code><a href='https://www.rdocumentation.org/packages/base/topics/integer'>integer(1)</a></code>.
Name or position of <code>DESeqResults</code>.</p></td>
    </tr>
    <tr>
      <th>lfcShrink</th>
      <td><p><code><a href='https://www.rdocumentation.org/packages/base/topics/logical'>logical(1)</a></code>.
Use shrunken log2 fold change (LFC) values.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments.</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p>the significance level for thresholding adjusted p-values</p></td>
    </tr>
    <tr>
      <th>main</th>
      <td><p>optional title for the plot</p></td>
    </tr>
    <tr>
      <th>xlab</th>
      <td><p>optional defaults to "mean of normalized counts"</p></td>
    </tr>
    <tr>
      <th>ylim</th>
      <td><p>optional y limits</p></td>
    </tr>
    <tr>
      <th>MLE</th>
      <td><p>if <code>betaPrior=TRUE</code> was used,
whether to plot the MLE (unshrunken estimates), defaults to FALSE.
Requires that <code><a href='results.html'>results</a></code> was run with <code>addMLE=TRUE</code>.
Note that the MLE will be plotted regardless of this argument,
if DESeq() was run with <code>betaPrior=FALSE</code>. See <code>lfcShrink</code>
for examples on how to plot shrunken log2 fold changes.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><code>ggplot</code>.</p>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>An MA plot is an application of a Bland–Altman plot for visual representation
of genomic data. The plot visualizes the differences between measurements
taken in two samples, by transforming the data onto M (log ratio) and A
(mean average) scales, then plotting these values.</p>
    
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>We are not allowing post hoc <code>alpha</code> or <code>lfcThreshold</code> cutoffs here.</p>
    
    <h2 class="hasAnchor" id="plotma-aliases"><a class="anchor" href="#plotma-aliases"></a>plotMA2 aliases</h2>

    
    <p>Aliased methods for original <code><a href='https://www.rdocumentation.org/packages/DESeq2/topics/plotMA'>DESeq2::plotMA()</a></code> S4 methods, which us
geneplotter instead of ggplot2. I prefer using ggplot2 instead, so the
primary methods defined here in the package mask DESeq2.</p>
    
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='https://www.rdocumentation.org/packages/DESeq2/topics/plotMA'>DESeq2::plotMA()</a></code>.</p></div>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/data'>data</a></span>(<span class='no'>deseq</span>)

<span class='co'>## Get genes from DESeqDataSet.</span>
<span class='no'>dds</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='as.html'>as</a></span>(<span class='no'>deseq</span>, <span class='st'>"DESeqDataSet"</span>)
<span class='no'>genes</span> <span class='kw'>&lt;-</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/utils/topics/head'>head</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/colnames'>rownames</a></span>(<span class='no'>dds</span>))
<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/print'>print</a></span>(<span class='no'>genes</span>)</div><div class='output co'>#&gt; [1] "gene001" "gene002" "gene003" "gene004" "gene005" "gene006"</div><div class='input'>
<span class='co'>## DESeqAnalysis ====</span>
<span class='fu'>plotMA</span>(<span class='no'>deseq</span>, <span class='kw'>results</span> <span class='kw'>=</span> <span class='fl'>1L</span>)</div><div class='output co'>#&gt; <span class='message'>condition_B_vs_A (shrunken LFC)</span></div><div class='img'><img src='plotMA-1.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Customize the colors.</span>
<span class='fu'>plotMA</span>(
    <span class='kw'>object</span> <span class='kw'>=</span> <span class='no'>deseq</span>,
    <span class='kw'>results</span> <span class='kw'>=</span> <span class='fl'>1L</span>,
    <span class='kw'>pointColor</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(
        <span class='kw'>downregulated</span> <span class='kw'>=</span> <span class='st'>"red"</span>,
        <span class='kw'>nonsignificant</span> <span class='kw'>=</span> <span class='st'>"black"</span>,
        <span class='kw'>upregulated</span> <span class='kw'>=</span> <span class='st'>"green"</span>
    )
)</div><div class='output co'>#&gt; <span class='message'>condition_B_vs_A (shrunken LFC)</span></div><div class='img'><img src='plotMA-2.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Directional support (up or down).</span>
<span class='fu'>plotMA</span>(<span class='no'>deseq</span>, <span class='kw'>results</span> <span class='kw'>=</span> <span class='fl'>1L</span>, <span class='kw'>direction</span> <span class='kw'>=</span> <span class='st'>"up"</span>, <span class='kw'>ntop</span> <span class='kw'>=</span> <span class='fl'>5L</span>)</div><div class='output co'>#&gt; <span class='message'>condition_B_vs_A (shrunken LFC)</span></div><div class='img'><img src='plotMA-3.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'>plotMA</span>(<span class='no'>deseq</span>, <span class='kw'>results</span> <span class='kw'>=</span> <span class='fl'>1L</span>, <span class='kw'>direction</span> <span class='kw'>=</span> <span class='st'>"down"</span>, <span class='kw'>ntop</span> <span class='kw'>=</span> <span class='fl'>5L</span>)</div><div class='output co'>#&gt; <span class='message'>condition_B_vs_A (shrunken LFC)</span></div><div class='img'><img src='plotMA-4.png' alt='' width='700' height='433' /></div><div class='input'>
<span class='co'>## Label genes manually.</span>
<span class='co'>## Note that either gene IDs or names (symbols) are supported.</span>
<span class='fu'>plotMA</span>(<span class='no'>deseq</span>, <span class='kw'>results</span> <span class='kw'>=</span> <span class='fl'>1L</span>, <span class='kw'>genes</span> <span class='kw'>=</span> <span class='no'>genes</span>)</div><div class='output co'>#&gt; <span class='message'>condition_B_vs_A (shrunken LFC)</span></div><div class='img'><img src='plotMA-5.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#details">Details</a></li>

      <li><a href="#note">Note</a></li>

      <li><a href="#plotma-aliases">plotMA2 aliases</a></li>

      <li><a href="#see-also">See also</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Michael Steinbaugh, Rory Kirchner</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://acidgenomics.com/'>Acid Genomics</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

