<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Differentially expressed gene heatmap — plotDEGHeatmap • DESeqAnalysis</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



  <link href="../extra.css" rel="stylesheet">
  

<meta property="og:title" content="Differentially expressed gene heatmap — plotDEGHeatmap" />
<meta property="og:description" content="This function is an extension of plotHeatmap() that is optimized for
automatic handling differentially expressed genes, rather than requiring
manual input of a gene vector. All of the optional parameters for
plotHeatmap() are also supported by this function." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DESeqAnalysis</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.10</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://acidgenomics.com/">
    <span class="fas fa fas fa-vial fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/acidgenomics/DESeqAnalysis/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Differentially expressed gene heatmap</h1>
    <small class="dont-index">Source: <a href='https://github.com/acidgenomics/DESeqAnalysis/blob/master/R/plotDEGHeatmap-methods.R'><code>R/plotDEGHeatmap-methods.R</code></a></small>
    <div class="hidden name"><code>plotDEGHeatmap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function is an extension of <code><a href='https://acidgenerics.acidgenomics.com/reference/plotHeatmap.html'>plotHeatmap()</a></code> that is optimized for
automatic handling differentially expressed genes, rather than requiring
manual input of a gene vector. All of the optional parameters for
<code><a href='https://acidgenerics.acidgenomics.com/reference/plotHeatmap.html'>plotHeatmap()</a></code> are also supported by this function.</p>
    </div>

    <pre class="usage"><span class='fu'>plotDEGHeatmap</span><span class='op'>(</span><span class='va'>object</span>, <span class='va'>...</span><span class='op'>)</span>

<span class='co'># S4 method for DESeqResults</span>
<span class='fu'>plotDEGHeatmap</span><span class='op'>(</span>
  <span class='va'>object</span>,
  <span class='va'>DESeqTransform</span>,
  alphaThreshold <span class='op'>=</span> <span class='cn'>NULL</span>,
  lfcThreshold <span class='op'>=</span> <span class='cn'>NULL</span>,
  baseMeanThreshold <span class='op'>=</span> <span class='cn'>NULL</span>,
  direction <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"both"</span>, <span class='st'>"up"</span>, <span class='st'>"down"</span><span class='op'>)</span>,
  title <span class='op'>=</span> <span class='cn'>TRUE</span>,
  subtitle <span class='op'>=</span> <span class='cn'>TRUE</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='co'># S4 method for DESeqAnalysis</span>
<span class='fu'>plotDEGHeatmap</span><span class='op'>(</span><span class='va'>object</span>, <span class='va'>i</span>, contrastSamples <span class='op'>=</span> <span class='cn'>FALSE</span>, <span class='va'>...</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>object</th>
      <td><p>Object.</p></td>
    </tr>
    <tr>
      <th>DESeqTransform</th>
      <td><p><code>DESeqTransform</code>.</p></td>
    </tr>
    <tr>
      <th>alphaThreshold</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code> or <code>NULL</code>.
Adjusted P value ("alpha") cutoff. If left <code>NULL</code>, will use the cutoff
defined in the object.</p></td>
    </tr>
    <tr>
      <th>lfcThreshold</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code> or <code>NULL</code>.
Log (base 2) fold change ratio cutoff threshold.
If left <code>NULL</code>, will use the cutoff defined in the object.</p></td>
    </tr>
    <tr>
      <th>baseMeanThreshold</th>
      <td><p><code><a href='https://rdrr.io/r/base/numeric.html'>numeric(1)</a></code> or <code>NULL</code>.
Base mean (i.e. average expression across all samples) threshold.
If left <code>NULL</code>, will use the cutoff defined in the object.
Applies in general to DESeq2 RNA-seq differential expression output.</p></td>
    </tr>
    <tr>
      <th>direction</th>
      <td><p><code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Include <code>"both"</code>, <code>"up"</code>, or <code>"down"</code> directions.</p></td>
    </tr>
    <tr>
      <th>title</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>, <code><a href='https://rdrr.io/r/base/character.html'>character(1)</a></code>.
Include contrast name as title?
Can manually define as <code>character</code>.</p></td>
    </tr>
    <tr>
      <th>subtitle</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
Include subtitle containing DEG information?</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Additional arguments.</p></td>
    </tr>
    <tr>
      <th>i</th>
      <td><p>Indices specifying elements to extract or replace. Indices are <code>numeric</code> or
<code>character</code> vectors, empty (<code>missing</code>), or <code>NULL</code>.</p>
<p>For more information:</p><pre><span class='fu'><a href='https://rdrr.io/r/utils/help.html'>help</a></span><span class='op'>(</span>topic <span class='op'>=</span> <span class='st'>"Extract"</span>, package <span class='op'>=</span> <span class='st'>"base"</span><span class='op'>)</span>
</pre></td>
    </tr>
    <tr>
      <th>contrastSamples</th>
      <td><p><code><a href='https://rdrr.io/r/base/logical.html'>logical(1)</a></code>.
<strong>Experimental</strong>. Only include the samples used to define the contrast
passed to <code><a href='https://rdrr.io/pkg/DESeq2/man/results.html'>DESeq2::results()</a></code>. This setting will break for complex DESeq2
contrasts (e.g. interaction effect).</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Plot.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>To adjust the annotation columns, modify the
<code><a href='https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html'>colData()</a></code> of the <code>counts</code> argument, which
must contain a <code>SummarizedExperiment</code>.</p>
    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    
<ul>
<li><p><code>plotDEGHeatmap,DESeqResults-method</code>: Passes to <code><a href='plotHeatmap.html'>plotHeatmap()</a></code> <code>SummarizedExperiment</code>
method defined in AcidPlots.</p></li>
<li><p><code>plotDEGHeatmap,DESeqAnalysis-method</code>: Passes to <code>DESeqResults</code> method.</p></li>
</ul>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>Updated 2020-08-05.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>deseq</span><span class='op'>)</span>

<span class='co'>## DESeqAnalysis ====</span>
<span class='fu'>plotDEGHeatmap</span><span class='op'>(</span><span class='va'>deseq</span>, i <span class='op'>=</span> <span class='fl'>1L</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='error'>Error in plotDEGHeatmap(deseq, i = 1L): could not find function "plotDEGHeatmap"</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by <a href='https://steinbaugh.com/'>Michael Steinbaugh</a>, <a href='https://acidgenomics.com/'>Acid Genomics</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


